tests/passes/compile-invoke/invoke-ref.expect

```diff
@@ -22,7 +22,6 @@ component main(@go go: 1, @clk clk: 1, @reset reset: 1) -> (@done done: 1) {
     group invoke0 {
       r0.in = f.r_in;
       r0.write_en = f.r_write_en;
-      f.r_out = r0.out;
```

Comment:
Why is this assignment being removed? For a `ref`-cell, all ports should be connected up right?

Reply:
Looking at the [original program](https://github.com/calyxir/calyx/blob/b6aa6a68be1db03a2a9de04945d7167581ea2d44/tests/passes/compile-invoke/invoke-ref.futil), `foo` never uses `r.out`, and so there's no reason to ever drive a signal with `r.out`.

If we add something like
`foo_out = r.out` in a group/continuous assignment within `foo`, then the pass will generate `f.r_out = r0.out;`.

---

Here's a full program we can test this on 

```
// -p validate -p compile-invoke
import "primitives/compile.futil";

component foo() -> (foo_out : 32) {
    cells {
        ref r = std_reg(32);
    }
    wires {
        foo_out = r.out; //this was added, along with the `foo_out` port
    }
    control {
        invoke r(in = 32'd10)();
    }
}

component main() -> () {
    cells {
        r0 = std_reg(32);
        f = foo();
    }
    wires {}
    control {
        invoke f[r=r0]()();
    }
}
```

Reply:
Hm, the thing I'm finding weird here is that the `invoke` statement should not know whether or not `r.out` is used in the rest of the program so why does it eliminate it? It should be the responsibility of another pass to clean up unused assignments right?

Reply:
I might be missing something, but to me this seems fine?
The original pass changes the signature of `foo` to this:
```
component foo(@go go: 1, @clk clk: 1, @reset reset: 1, r_out: 32, r_done: 1) -> (@done done: 1, r_in: 32, r_write_en: 1) {
```
And wires up `f.r_out = r0.out` as part of the `invoke` group.

But we are guaranteed

Reply:
I'm hoping to get this merged because there are other PRs I'm working on that build off this one, but want to confirm that this makes sense. Could you take a look at this again/could we discuss @rachitnigam?
cc @sampsyo as another person who might have context and input on this

Reply:
Okay, let's go ahead and merge this then!